### PR TITLE
Reformat Input Context structure

### PR DESCRIPTION
**Pull request by ankitjena**
## Changes
- Reformat the context to be reused by graphback packages #83 
### Structure of the Context
```
export interface FieldContext {
  name: string
  type: string
  isArray: boolean
  isType: boolean
  isNull: boolean
  directives: object
  hasDirectives: boolean
}

export interface InputContext {
  name: string
  fields: FieldContext[]
  config: object
}
```
`config` is an object which has the following structure
```
{
  paginate: boolean
  create: boolean,
  update: boolean,
  delete: boolean,
  find: boolean,
  findAll: boolean
}
```
By default the config is 
```
{
  paginate: false,
  create: true,
  update: true,
  delete: true,
  find: true,
  findAll: true
}
```

### CODE REVIEW

File: packages/graphback-schema/src/plugin.ts [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
+import { buildTargetContext, TargetContext } from './targetType';
 
 export const plugin = (schema: GraphQLSchema) => {
-  const printedSchema: string = printSchema(schema)
-  const astNode: DocumentNode = parse(printedSchema)
-  
-  const inputTypes = visit(astNode, { leave: inputTypeVisitor })
 
-  const context: TargetContext = buildTargetContext(inputTypes.definitions.reverse())
+  const inputContext = new GraphbackContextCreator(printSchema(schema)).createInputContext()
```
Comment (ankitjena) [inline packages/graphback-schema/src/plugin.ts]: We are initializing the Input Context here. Current way using visitor pattern doesn't require using codegen plugin as we can pass this Input Context to different use cases. 

Codegen further needs to accept schema instance everytime which is really not required.

@wtrocki  WDYT?

File: packages/graphback/src/InputTypeVisitor.ts [hunk 2]
```diff
@@ -1,7 +1,16 @@
-import { DirectiveDefinitionNode, FieldDefinitionNode, ListTypeNode, NamedTypeNode, NameNode, NonNullTypeNode, ObjectTypeDefinitionNode } from 'graphql';
+import { ArgumentNode, BooleanValueNode, DirectiveDefinitionNode, DirectiveNode, FieldDefinitionNode, ListTypeNode, NamedTypeNode, NameNode, NonNullTypeNode, ObjectTypeDefinitionNode, StringValueNode } from 'graphql';
 
 const scalars = ['ID', 'Int', 'Float', 'String', 'Boolean']
 
+const defaultConfig = {
```
Comment (wtrocki) [inline packages/graphback/src/InputTypeVisitor.ts]: Quick question. Do we still use directives here and override this values?

File: packages/graphback/src/directives.ts [hunk 3]
```diff
@@ -1,7 +1,12 @@
-const directives = `
-  directive @OneToOne(field: String) on FIELD
-  directive @OneToMany(field: String) on FIELD
-  directive @ManyToMany(tablename: String) on FIELD
+const directives = `directive @OnetoMany(field: String) on FIELD_DEFINITION
+directive @OneToOne(field: String) on FIELD_DEFINITION
+directive @ManyToMany(tablename: String) on FIELD_DEFINITION
+directive @config(paginate: Boolean,
```
Comment (wtrocki) [inline packages/graphback/src/directives.ts]: I will suggest using `Generator` or `Model`. Let's use capitals maybe.

**Comment by wtrocki** [comment]
@ankitjena What is the plan here? Are we planning to conform to this config in this PR and do resolves or there will be another PR for that? I would love to have pagination enabled by default (sorry :)).

Can I verify that or it is still in the works?

File: packages/graphback/src/InputTypeVisitor.ts (hunk 2 above)
Comment (ankitjena) [reply packages/graphback/src/InputTypeVisitor.ts]: Yes we do that while visiting the nodes.

**Comment by ankitjena** [comment]
> @ankitjena What is the plan here? Are we planning to conform to this config in this PR and do resolves or there will be another PR for that? I would love to have pagination enabled by default (sorry :)).
> 

Yes, will do resolvers in separate PR.

**Comment by wtrocki** [comment]
Let's merge that and work on new aproach

**Pull request merged**